In [1]:
import pandas
import numpy as np

# Machine Learning - IN221 - February 2025
## Eng. Ahmed Métwalli
## Week 2 - Hands-on Coding

<img src = 'Images/datasc-660x434.png'>

## [View Section: Linear Regression Guidelines](SecWeek2.pdf)

# Getting started with Data Scient Lifecycle coding

## Importing Packages

In [5]:
# Pandas > Data manipulation
# Import Numpy (Numerical operations)
# Matplotlib for visualization
import pandas as pd,\
        numpy as np,\
        matplotlib.pyplot as plt

## Importing Covid-19 Dataset

In [8]:
# Filename
file_name = "Public Datasets/covid_19_clean_complete.csv.zip" # The file of the dataset

In [9]:
df = pd.read_csv(file_name, compression='zip') # The dataframe to be imported

## Data & Business Understanding

### Objective: Our aim is to predict the amount of death in the next days or weeks.

In [12]:
df.info() # To view information about each attribute

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49068 entries, 0 to 49067
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Province/State  14664 non-null  object 
 1   Country/Region  49068 non-null  object 
 2   Lat             49068 non-null  float64
 3   Long            49068 non-null  float64
 4   Date            49068 non-null  object 
 5   Confirmed       49068 non-null  int64  
 6   Deaths          49068 non-null  int64  
 7   Recovered       49068 non-null  int64  
 8   Active          49068 non-null  int64  
 9   WHO Region      49068 non-null  object 
dtypes: float64(2), int64(4), object(4)
memory usage: 3.7+ MB


In [24]:
df['Province/State'].isna().sum()

34404

In [29]:
df.drop('Province/State', axis=1, inplace=True)

In [31]:
df.describe()

Lat          Long     Confirmed         Deaths     Recovered  \
count  49068.000000  49068.000000  4.906800e+04   49068.000000  4.906800e+04   
mean      21.433730     23.528236  1.688490e+04     884.179160  7.915713e+03   
std       24.950320     70.442740  1.273002e+05    6313.584411  5.480092e+04   
min      -51.796300   -135.000000  0.000000e+00       0.000000  0.000000e+00   
25%        7.873054    -15.310100  4.000000e+00       0.000000  0.000000e+00   
50%       23.634500     21.745300  1.680000e+02       2.000000  2.900000e+01   
75%       41.204380     80.771797  1.518250e+03      30.000000  6.660000e+02   
max       71.706900    178.065000  4.290259e+06  148011.000000  1.846641e+06   

             Active  
count  4.906800e+04  
mean   8.085012e+03  
std    7.625890e+04  
min   -1.400000e+01  
25%    0.000000e+00  
50%    2.600000e+01  
75%    6.060000e+02  
max    2.816444e+06

In [13]:
df.head() # To view/display the first 5 records of the data

Province/State Country/Region       Lat       Long        Date  Confirmed  \
0            NaN    Afghanistan  33.93911  67.709953  2020-01-22          0   
1            NaN        Albania  41.15330  20.168300  2020-01-22          0   
2            NaN        Algeria  28.03390   1.659600  2020-01-22          0   
3            NaN        Andorra  42.50630   1.521800  2020-01-22          0   
4            NaN         Angola -11.20270  17.873900  2020-01-22          0   

   Deaths  Recovered  Active             WHO Region  
0       0          0       0  Eastern Mediterranean  
1       0          0       0                 Europe  
2       0          0       0                 Africa  
3       0          0       0                 Europe  
4       0          0       0                 Africa

In [24]:
df.describe()

Lat          Long     Confirmed         Deaths     Recovered  \
count  49068.000000  49068.000000  4.906800e+04   49068.000000  4.906800e+04   
mean      21.433730     23.528236  1.688490e+04     884.179160  7.915713e+03   
std       24.950320     70.442740  1.273002e+05    6313.584411  5.480092e+04   
min      -51.796300   -135.000000  0.000000e+00       0.000000  0.000000e+00   
25%        7.873054    -15.310100  4.000000e+00       0.000000  0.000000e+00   
50%       23.634500     21.745300  1.680000e+02       2.000000  2.900000e+01   
75%       41.204380     80.771797  1.518250e+03      30.000000  6.660000e+02   
max       71.706900    178.065000  4.290259e+06  148011.000000  1.846641e+06   

             Active  
count  4.906800e+04  
mean   8.085012e+03  
std    7.625890e+04  
min   -1.400000e+01  
25%    0.000000e+00  
50%    2.600000e+01  
75%    6.060000e+02  
max    2.816444e+06

## Data Preparation

In [ ]:
df.isna().sum() # View the number of missing values

Province/State    34404
Country/Region        0
Lat                   0
Long                  0
Date                  0
Confirmed             0
Deaths                0
Recovered             0
Active                0
WHO Region            0
dtype: int64

### Documenting current state

Province/State contains many nulls 34404 out 49068, therefore we can drop it or dive deeper to understand the existing non-missing values.

In [22]:
df.drop('Province/State',axis=1,inplace=True)

In [23]:
df

Country/Region        Lat       Long        Date  Confirmed  \
0                Afghanistan  33.939110  67.709953  2020-01-22          0   
1                    Albania  41.153300  20.168300  2020-01-22          0   
2                    Algeria  28.033900   1.659600  2020-01-22          0   
3                    Andorra  42.506300   1.521800  2020-01-22          0   
4                     Angola -11.202700  17.873900  2020-01-22          0   
...                      ...        ...        ...         ...        ...   
49063  Sao Tome and Principe   0.186400   6.613100  2020-07-27        865   
49064                  Yemen  15.552727  48.516388  2020-07-27       1691   
49065                Comoros -11.645500  43.333300  2020-07-27        354   
49066             Tajikistan  38.861000  71.276100  2020-07-27       7235   
49067                Lesotho -29.610000  28.233600  2020-07-27        505   

       Deaths  Recovered  Active             WHO Region  
0           0          0       0  Eastern Mediterranean  
1           0          0       0                 Europe  
2           0          0       0                 Africa  
3           0          0       0                 Europe  
4           0          0       0                 Africa  
...       ...        ...     ...                    ...  
49063      14        734     117                 Africa  
49064     483        833     375  Eastern Mediterranean  
49065       7        328      19                 Africa  
49066      60       6028    1147                 Europe  
49067      12        128     365                 Africa  

[49068 rows x 9 columns]